### PR TITLE
JS docs

### PR DESCRIPTION
**Pull request by gbirke**
Added some JS docstrings

### CODE REVIEW

File: DeepCat.js [hunk 1]
```diff
@@ -149,6 +155,13 @@
 		$( '#searchform' ).submit();
 	}
 
+	/**
+	 * Replace "deepcat:" search terms with "incategory:" terms from DeepCat response
+	 *
+	 * @param {array} responses Category search results
+	 * @param {array} newSearchTerms Original search terms provided by the user
+	 * @returns {array} Modified newSearchTerms
```
Comment (wmde-manicki) [inline DeepCat.js]: Although PhpStorm inserts `@returns` tag, the tag we are using is actually `@return`, see: https://www.mediawiki.org/wiki/Manual:Coding_conventions/JavaScript#Tags

**Comment by wmde-manicki** [comment]
You are using `{array}` as a type of parameters and return values here.
Arrays are not Javascript's primitives, so according to https://www.mediawiki.org/wiki/Manual:Coding_conventions/JavaScript#Types and https://github.com/senchalabs/jsduck/wiki/Type-Definitions#type-names it should rather be `{Array}`.

BUT: I see that javadocs for other parts of code use `@return {string[]}`. The methods you've added the docs to are also processing string only (afaik), so it could also fit for the docs in this pull request.

I have no idea which convention is more correct here, `@return {Array}` or `@return {string[]}`. Any comments on that @thiemowmde @WMDE-Fisch ?

**Comment by thiemowmde** [comment]
Try to be as specific as possible. Common practice is to either use `{string[]}` if all elements in the array are guaranteed to have the same type, or `{Array}` if the array contains mixed stuff.

**Comment by wmde-manicki** [comment]
Thanks a lot @thiemowmde!
It makes definitely sense to do it exactly that way, we'll stick to that.

**Comment by wmde-manicki** [comment]
Updated the types accordingly and rebased on top of current master.

**Pull request merged**